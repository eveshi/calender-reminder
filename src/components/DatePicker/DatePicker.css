*,
*::after,
*::before {
  margin: 0;
  padding: 0;
  box-sizing: border-box; }

html {
  font-size: 62.5%; }
  @media only screen and (max-width: 400px) {
    html {
      font-size: 50%; } }

body {
  font-family: 'Roboto', sans-serif;
  background-color: #F5F5F4; }

a svg {
  height: 4rem;
  width: 4rem;
  text-decoration: none;
  fill: #5F513F; }
  a svg:hover {
    background-color: transparent; }

@font-face {
  font-family: 'Roboto', sans-serif;
  src: url("https://fonts.googleapis.com/css?family=Roboto"); }

.DatePicker {
  display: flex; }
  .DatePicker:focus {
    outline: none; }
  .DatePicker .Button {
    display: flex;
    justify-content: space-around; }
    .DatePicker .Button:hover {
      background-color: transparent;
      text-shadow: 2px 0 2px #ffffff;
      transform: scale(0.98); }
  .DatePicker_date {
    color: #5F513F;
    margin: auto .5rem;
    font-size: 2.4rem; }
  .DatePicker_picker {
    display: flex;
    flex-direction: row;
    align-items: center; }
    .DatePicker_picker_select {
      font-size: 2.4rem;
      margin: auto .5rem;
      background-color: transparent;
      border: none;
      color: #5F513F; }
      .DatePicker_picker_select:focus {
        outline: none; }
    .DatePicker_picker .Button {
      border: 2px solid #BD924B;
      border-radius: 1rem;
      height: 3rem;
      font-size: 1.4rem;
      padding: 0 1rem;
      box-shadow: 2px 1px 1px #A5A4A2;
      margin-left: 1rem; }
