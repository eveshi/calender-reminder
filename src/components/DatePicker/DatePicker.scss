@import '../../stylesheets/main.scss';

.DatePicker {
  display: flex;

  &:focus {
    outline: none;
  }

  .Button {
    display: flex;
    justify-content: space-around;

    &:hover{
      background-color: transparent;
      text-shadow: 2px 0 2px #ffffff;
      transform: scale(0.98);
    }
  }

  &_date {
    color: $color-secondary-darker;
    margin: auto .5rem;
    font-size: 2.4rem;
  }

  &_picker {
    display: flex;
    flex-direction: row;
    align-items: center;

    &_select {
      font-size: 2.4rem;
      margin: auto .5rem;
      background-color: transparent;
      border: none;
      color: $color-secondary-darker;

      &:focus {
        outline: none;
      }
    }

    .Button {
      border: 2px solid $color-secondary-dark;
      border-radius: 1rem;
      height: 3rem;
      font-size: 1.4rem;
      padding: 0 1rem;
      box-shadow: 2px 1px 1px $color-primary-middle;
      margin-left: 1rem;
    }
  }
}