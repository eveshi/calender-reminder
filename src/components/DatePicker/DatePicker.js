import React, { PureComponent } from 'react';
import { connect } from 'react-redux';
import PropTypes from 'prop-types';

import * as utility from '../../utility/utility';
import Button from '../Button/Button';
import * as actions from '../../store/actions';

import YearPicker from './Pickers/YearPicker';
import MonthPicker from './Pickers/MonthPicker';
import DayPicker from './Pickers/DayPicker';

import './DatePicker.css';

export class DatePicker extends PureComponent {
  state = {
    datePickerDisabled: true,
  }

  componentDidMount() {
    const { datePickerDisabled } = this.props;
    this.setState({
      datePickerDisabled,
    });
  }

  dateChangeHandler = (event, selector) => {
    const {
      changeSessionDay,
      changeSessionMonth,
      changeSessionYear,
    } = this.props;
    const value = parseInt(event.target.value, 10);

    if (selector === 'year') {
      changeSessionYear(value);
    }

    if (selector === 'month') {
      changeSessionMonth(value);
    }

    if (selector === 'day') {
      changeSessionDay(value);
    }
  };

  datePickerDisabledHandler = () => {
    const { datePickerDisabled } = this.state;
    this.setState({
      datePickerDisabled: !datePickerDisabled,
    });
  };

  applyNewDate = () => {
    const {
      yearPicked,
      monthPicked,
      dayPicked,
      changeDay,
      changeYearAndMonth,
    } = this.props;
    changeYearAndMonth(yearPicked, monthPicked);
    changeDay(dayPicked);

    const { datePickerDisabledHandler } = this;
    datePickerDisabledHandler();
  };

  render() {
    const {
      yearPicked,
      monthPicked,
      dayPicked,
      yearArray,
      monthArray,
      dayArray,
      dayPickerDisabled,
      applyButtonDisabled,
    } = this.props;

    const {
      dateChangeHandler,
      datePickerDisabledHandler,
      applyNewDate,
    } = this;

    const { datePickerDisabled } = this.state;

    const dayPicker = dayPickerDisabled === true
      ? null
      : (
        <DayPicker
          dayArray={dayArray}
          dayPicked={dayPicked}
          onChange={event => dateChangeHandler(event, 'day')}
        />
      );

    const dayDisplay = dayPickerDisabled === true
      ? null
      : (
        <p className="DatePicker_date">{dayPicked}</p>
      );

    const applyButtonDisplay = applyButtonDisabled === true
      ? null
      : (
        <Button
          onClick={() => applyNewDate()}
        >
          Apply
        </Button>
      );

    const dateDisplay = datePickerDisabled === true
      ? (
        <Button onClick={() => datePickerDisabledHandler()}>
          <p className="DatePicker_date">
            {yearPicked.toString()}
          </p>
          <p className="DatePicker_date">
            {utility.monthStringGenerator(monthPicked)}
          </p>
          {dayDisplay}
          <p className="DatePicker_date">
            &gt;
          </p>
        </Button>
      )
      : (
        <div className="DatePicker_picker">
          <form>
            <YearPicker
              yearPicked={yearPicked}
              yearArray={yearArray}
              onChange={event => dateChangeHandler(event, 'year')}
            />
            <MonthPicker
              monthPicked={monthPicked}
              monthArray={monthArray}
              onChange={event => dateChangeHandler(event, 'month')}
            />
            {dayPicker}
          </form>
          {applyButtonDisplay}
        </div>
      );

    return (
      <div className="DatePicker">
        {dateDisplay}
      </div>
    );
  }
}

DatePicker.propTypes = {
  yearPicked: PropTypes.number.isRequired,
  monthPicked: PropTypes.number.isRequired,
  dayPicked: PropTypes.number.isRequired,
  yearArray: PropTypes.arrayOf(PropTypes.number).isRequired,
  monthArray: PropTypes.arrayOf(PropTypes.object).isRequired,
  dayArray: PropTypes.arrayOf(PropTypes.number).isRequired,
  changeDay: PropTypes.func.isRequired,
  changeYearAndMonth: PropTypes.func.isRequired,
  datePickerDisabled: PropTypes.bool,
  dayPickerDisabled: PropTypes.bool,
  applyButtonDisabled: PropTypes.bool,
  changeSessionYear: PropTypes.func.isRequired,
  changeSessionMonth: PropTypes.func.isRequired,
  changeSessionDay: PropTypes.func.isRequired,
};

DatePicker.defaultProps = {
  dayPickerDisabled: true,
  datePickerDisabled: true,
  applyButtonDisabled: true,
};

const mapStateToProps = state => ({
  yearPicked: state.sessionState.year,
  monthPicked: state.sessionState.month.index,
  dayPicked: state.sessionState.day,
  yearArray: state.sessionState.yearArray,
  monthArray: state.sessionState.monthArray,
  dayArray: state.sessionState.dayArray,
});

const mapActionToProps = dispatch => ({
  changeDay: day => dispatch(actions.changeDay(day)),
  changeYearAndMonth: (year, month) => dispatch(actions.changeYearAndMonth(year, month)),
  changeSessionDay: day => dispatch(actions.changeSessionDay(day)),
  changeSessionMonth: month => dispatch(actions.changeSessionMonth(month)),
  changeSessionYear: year => dispatch(actions.changeSessionYear(year)),
});

export default connect(mapStateToProps, mapActionToProps)(DatePicker);
