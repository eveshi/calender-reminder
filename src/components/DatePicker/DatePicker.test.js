import React from 'react';
import renderer from 'react-test-renderer';
import { shallow, configure, mount } from 'enzyme';
import Adapter from 'enzyme-adapter-react-16';
import { DatePicker } from './DatePicker';

configure({ adapter: new Adapter() });

const props = {
  yearInit: Math.random().toFixed(4)*4,
  monthInit: Math.random().toFixed(2)*2,
  dayInit: Math.random().toFixed(2)*4,
  changeDay: jest.fn(),
  changeYearAndMonth: jest.fn(),
}

describe('snapshot', () => {
  it('renders correctly', () => {
    const tree = renderer.create(<DatePicker {...props}/>).toJSON();
    expect(tree).toMatchSnapshot();
  });
});