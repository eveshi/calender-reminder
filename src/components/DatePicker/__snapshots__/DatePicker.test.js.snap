// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`snapshot renders correctly 1`] = `
<div
  className="DatePicker"
>
  <button
    className="Button"
    disabled={null}
    id={null}
    onClick={[Function]}
    type="button"
  >
    <p
      className="DatePicker_date"
    >
      2032
    </p>
    <p
      className="DatePicker_date"
    >
      Jul
    </p>
    <p
      className="DatePicker_date"
    >
      &gt;
    </p>
  </button>
</div>
`;
